src/utils/db.ts

```diff
@@ -467,6 +476,141 @@ class DB {
+
+        // await sendCreateGroupJob(groupName, user1Email)
+      } catch (error) {
+        console.error(`Something went wrong: ${error}`);
+        reject(error);
+      } finally {
+        await session.close();
+        resolve(results);
+      }
+    })
+  }
+
+  // Creates a channel node and links it the the parent group
+  async deleteChannel(channelUuid: string, groupId: Integer) {
+    const session = this.localDriver.session({ database: "neo4j" });
+    let results = [];
+    return new Promise(async (resolve, reject) => {
+      try {
+        const writeQuery = `MATCH (c:Channel {groupId: $groupId, channelUuid: $channelUuid})-[:CHANNEL]->(g:Group {id: $groupId}) 
+        DETACH DELETE c`;
+
+        const writeResult = await session.executeWrite((tx) =>
+          tx.run(writeQuery, { groupId, channelUuid })
+        );
+        console.log("DELETED CHANNEL FOR: ", groupId, "<-", channelUuid);
+
+        // await sendCreateGroupJob(groupName, user1Email)
+      } catch (error) {
+        console.error(`Something went wrong: ${error}`);
+        reject(error);
+      } finally {
+        await session.close();
+        resolve(results);
+      }
+    })
+  }
+
+  // Saves when user joins a channel to neo4j, links user's node to channel when user joins
+  async joinChannel(userEmail: String, channelUuid: String) {
+    const session = this.localDriver.session({ database: "neo4j" });
+    let results = [];
+    return new Promise(async (resolve, reject) => {
+      console.log(userEmail, channelUuid)
+      try {
+        const writeQuery = `MATCH (u:User {email: $userEmail})
+        MATCH (c:Channel {channelUuid: $channelUuid})
+        MERGE (u)-[r:CHANNELMEMBER]->(c)
+        RETURN u, c`;
+
+        const writeResult = await session.executeWrite((tx) =>
+          tx.run(writeQuery, { userEmail, channelUuid })
+        );
+
+        console.log(writeResult);
+
+      } catch (err) {
+        reject(err);
+      } finally {
+        await session.close();
+        resolve(results);
+      }
+    })
+  }
+
+  // Removes user's link to channel when user leaves a channel
+  async leaveChannel(userEmail: String, channelUuid: String) {
+    const session = this.localDriver.session({ database: "neo4j" });
+    let results = [];
+    return new Promise(async (resolve, reject) => {
+      try {
+        const writeQuery = `MATCH (u:User {email: $userEmail})
+        MATCH (c:Channel {uuid: $channelUuid})
+        MATCH (u)-[r:CHANNELMEMBER]->(c)
+        DELETE r
+        RETURN u, c`;
+
+        const writeResult = await session.executeWrite((tx) =>
+        tx.run(writeQuery, { userEmail, channelUuid })
+        );
+
+      } catch (err) {
+        
+        reject(err);
+      } finally {
+        await session.close();
+        resolve(results);
+      }
+    })
+  }
+
+    // Removes user's link to channel when user leaves a channel
+    async leaveAllChannels(userEmail: String) {
```

Comment:
What is the use case for leaveChannel vs leaveAllChannels? Do we need both?

Reply:
I will remove leaveChannel, since leaveAllChannels accomplishes the same thing and covers more edge cases.